pkg/cli/admin/inspect/namespace.go

```diff
@@ -92,3 +109,58 @@ func (o *InspectOptions) gatherNamespaceData(baseDir, namespace string) error {
 	}
 	return nil
 }
+
+// podEligibleForCollection aims to filter out the daemonset pods running on
+// data planes if node count is greater than a threshold,
+// in order to save significant storage size and reduces the collection duration.
+// If these pods need to be collected, recommended path is
+// to use node-selector flag (i.e. --node-selector='node-role.kubernetes.io/worker').
+func (o *InspectOptions) podEligibleForCollection(pod *corev1.Pod, nodes []corev1.Node) (bool, error) {
+	isDaemonSet := false
+	if len(pod.OwnerReferences) > 0 {
+		for _, OwnerRef := range pod.OwnerReferences {
+			if OwnerRef.Kind == "DaemonSet" {
+				isDaemonSet = true
+				break
+			}
+		}
+	}
+	if !isDaemonSet {
+		return true, nil
+	}
+
+	workerNodeCount := 0
+	for _, node := range nodes {
+		if _, ok := node.Labels["node-role.kubernetes.io/control-plane"]; ok {
+			continue
+		}
+		workerNodeCount++
+	}
+	const daemonsetCollectionMaxNodeSize = 18
+	if workerNodeCount <= daemonsetCollectionMaxNodeSize {
+		return true, nil
+	}
+
+	var podNode *corev1.Node
+	for _, node := range nodes {
+		if node.Name != pod.Spec.NodeName {
```

Comment:
This tests whether a DS pod is scheduled to a non-control-plane node. Would it make more sense to do the opposite and check whether a DS pod is "not" scheduled to a control plane node? Usually the number of control-plane nodes is much lower than the non-control-plane node.

Reply:
Gotcha. Then the podEligibleForCollection's description needs to be updated to better phrase the "running on data planes" since the default value of the node selector is not known in advance.

Reply:
I've updated the description. Please let me know what you think. Thank you.